### PR TITLE
Rewrite/extend for discussion

### LINKED ISSUE #102: Fix typo
https://github.com/dev-academy-programme/curriculum/tree/master/concepts/problem-solving

First bullet under Capabilities should read: 
* persevering <del>perservering</del> with a tough problem

### PR DESCRIPTION
**Pull request by richchurcher**
This is rather a core document for the school, and I'd like it to be a process that everyone agrees on. I think it's a resource every student should be familiar with, as a system they should use automatically when they get stuck.

I'd welcome suggestions! Maybe if we leave it in PR for a little while and I can make revisions as requested.

fixes #102

### CODE REVIEW

**Comment by peterjacobson** [comment]
Looks good to me rich - I'd be happy to accept the PR, but may leave it here for others to comment on?

File: concepts/problem-solving/README.md [hunk 1]
```diff
@@ -1,96 +1,149 @@
...
-    1. Ask peers on cohort slack
-    2. Tag a coach
-4. Reflect
-
-**1. PseudoCode**    
+3. Simplify
+4. Read error message
+5. Improve visibility of problems using console output
+6. Debuggers: 
+  * Chrome, Firefox or Edge dev tools
```
Comment (locksmithdon) [inline concepts/problem-solving/README.md]: For completeness we should add Safari to this list. Its dev tools aren't bad and quite important when doing iOS Cordova apps.

File: concepts/problem-solving/README.md [hunk 2]
```diff
@@ -1,96 +1,149 @@
...
+If you've been problem solving for over 15 mins and you feel like you've hit a dead end, post a message in your cohort Slack channel. It's likely someone else has had and resolved a similar issue, or has the perspective, understanding or experience to help you resolve your issue.
 
+Students are sometimes reluctant to do this, for a variety of reasons:
+ * "I don't want to take up their time"
+ * "I don't them to think I'm not as smart as they are"
+ * "I'm ashamed of how long this is taking me"
+ * "I've already asked for help three times this week!"
 
-## Prerequisites
+It's human nature to want to be independent, but this attitude can sometimes slow you down, and in an industry context it can slow down an entire team. Consider, what if by remaining quiet you're denying another team member the opportunity to become a better teacher? What if the answer is something you couldn't possibly have known, but another developer has learned only through long and painful experience? What if someone else has the exact same problem and could benefit from your speaking up?
```
Comment (locksmithdon) [inline concepts/problem-solving/README.md]: I love this SO much!

File: concepts/problem-solving/README.md [hunk 3]
```diff
@@ -1,96 +1,149 @@
...
 
-None
+What if the team becomes stronger when we admit we don't know something?
```
Comment (locksmithdon) [inline concepts/problem-solving/README.md]: Such perspective!

File: concepts/problem-solving/README.md [hunk 4]
```diff
@@ -1,96 +1,149 @@
...
+  * What you _expect_ the code to do
+  * What it _actually_ does
+  * Any error messages you receive
+
+### 11. Reflect  
+
+Sometimes it's tempting to charge ahead, steamroller-like, once you've finally managed to kick that pesky problem to the curb. We suggest, first and foremost, you're probably due a break! Stand up, relax, stretch, breathe. And consider:
+ * Which approaches worked well?    
+ * What did I get stuck on?   
+ * Why did it take so long to resolve a silly bug? A common question!
```
Comment (locksmithdon) [inline concepts/problem-solving/README.md]: Could add: What were the troubleshooting strategies/patterns that helped (or could have helped) that I should consider for the future?

**Comment by locksmithdon** [comment]
I'm real happy with this too and support it being merged right away.

**Comment by richchurcher** [comment]
Committing for now so it can be easily linked to.

**Pull request merged**